code/modules/food_and_drinks/food/snacks_bread.dm

```diff
@@ -228,6 +228,11 @@
 		fried.forceMove(src)
 		trash = fried
 
+/obj/item/reagent_containers/food/snacks/deepfryholder/generate_trash(atom/location)
+	if(trash)
+		QDEL_NULL(trash)
```

Comment:
Trash is intentionaly set in init for non food items. This seems like it will cause problems with non food items that get fried

Reply:
got rid of trash var